ocaml/libs/clock/timer.ml

```diff
@@ -0,0 +1,53 @@
+type t = {start: Ptime.t; elapsed: Mtime_clock.counter; timeout: Mtime.Span.t}
+
+type remaining = Spare of Mtime.Span.t | Excess of Mtime.Span.t
```

Comment:
Spare and Excess go nicely together, but I was thinking that maybe we should call it 'Expired', I think that is the more usual terminology associated with timers (an expired timer).
But then what would we use instead of 'spare'?
(this also applies to my epoll PR)

Reply:
best before 😹

Reply:
I'm thinking about using
`countdown = Left | Expired`, but I'm unconvinced. Maybe while writing tests I can think of something that feels more intuitive

Reply:
how about Remaining and Expired?

Reply:
That should work, I'll try

> best before 😹

If we want to go the succinct way, we go for `Yet | Since`.